src/routers/semester.ts

```diff
@@ -1,24 +1,49 @@
 import express from 'express';
+import { create, read, update, del } from '../operators/semesterOperations';
 
 const semesterRouter = express.Router();
 
-semesterRouter.get('/', (req, res) => {
-	console.log('Get Semester');
-	res.send('Get semesterRouter');
+semesterRouter.get('/:id', async (req, res) => {
+	try {
+		console.log('Get Semester');
+		const semester = await read(req.params.id);
+		const resData = {
+			...semester,
+			courses: Array.from(semester?.courses || []),
+		};
+		res.send(resData);
+	} catch (err) {
+		res.status(404).send(err);
```

Comment:
`send("Semester not found.")` I suggest making the string a constant at the top of the file, and the same change is needed for all of the 404 responses per discussion at the meeting 😄